[Playback 25] Paywall Story

SergioEstevao opened the pull request:
| 📘 Part of: [Playback 25](https://linear.app/a8c/project/ios-playback-2025-7fed05fea613/overview)
|:---:|

Fixes PCIOS-186 <!-- issue number, if applicable -->

<!-- Please include a summary of what this PR is changing and why these changes are needed. -->
Implements the plus paywall story

<img width="320"  alt="Simulator Screenshot - iPhone 14 - 2025-11-07 at 21 43 25" src="https://github.com/user-attachments/assets/0c4b6866-eeb4-4739-84e6-c55f7a4db53e" />

**Note:** ~There is an issue with the video background color does not match with the screen background color~
The background color was fixed by using a video background that matches the video background color for the animation

## To test

- Start the app
- Login with an user that has some stats for 2025 and **doesn't have** a subscription
- Go to Profile -> Settings -> Beta features
- Enable the FF  `End of Year 2025`
- Go to Developer Menu and tap on "End Of Year -> Reset modal/profile badge" and select 2025
- Restart the app
- Check that if you see the prompt for the End of the Year on the Podcasts screen
- Tap on View My Playback 2025
- Go to Profile 
- Check that you see a banner for the playback 2025
- Tap on it
- Check if Playback 2025 screen shows correctly
- Skip the stories until you get to the plus paywall screen
- Check if video shows correctly and loops
- Check if text is correct.


## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

SergioEstevao committed:
Paywall story added

SergioEstevao committed:
Fix video alignment

SergioEstevao committed:
Loop the video

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/109">8.0</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

SergioEstevao committed:
Fix aspect ratio calculations

SergioEstevao committed:
Fix replay onAppear

SergioEstevao committed:
Use custom video player

This fix interaction issues and solves the layout issues that where adding black edges

SergioEstevao committed:
Don't show video background

danielebogo commented on podcasts/End of Year/Stories/2025/PaidStoryWallView2025.swift:
```diff
@@ -0,0 +1,95 @@
+}
+
+#Preview {
+    PaidStoryWallView2025(subscriptionTier: .plus)
```
@SergioEstevao is it possible to also have the preview for Patron as well? Just a copy with the different tier and add the label to the previews/

SergioEstevao replied:
Fixed here: https://github.com/Automattic/pocket-casts-ios/pull/3669/commits/3face90e4e199c342af6d0d054473886ca36adf9

SergioEstevao committed:
Fix video positioning and background

SergioEstevao committed:
Ensure text fits screen always

danielebogo commented on podcasts/End of Year/Stories/2025/PaidStoryWallView2025.swift:
```diff
@@ -0,0 +1,104 @@
+                }
+                .buttonStyle(BasicButtonStyle(textColor: .black, backgroundColor: Color.clear, borderColor: .black))
+                .padding(.horizontal, 24)
+                .padding(.vertical, 6)
```
The padding between text and button is bigger than(24). Considering the footer has 4 on each part in a vertical padding, we should add 20 here
```suggestion
                .padding(.top, 20)
                .padding(.bottom, 6)
```

SergioEstevao replied:
Fixed here: https://github.com/Automattic/pocket-casts-ios/pull/3669/commits/18e09f88bf4a5279dec644d9f2f558adafe55be6

danielebogo reviewed (changes requested):
nice @SergioEstevao ! The video animation looks really cool! Just a comment about the space between the footer text and button

SergioEstevao committed:
Update padding

SergioEstevao committed:
Add patron preview

SergioEstevao committed:
Fix lint issue

SergioEstevao commented:
@danielebogo ready for another look

danielebogo reviewed (approved):
:shipit:
